Build failure: playwright-driver.browsers

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

`playwright-driver.browser` recently broke (and therefore break my PR https://github.com/NixOS/nixpkgs/pull/425503/):
```
$ nix build .#playwright-driver.browsers
```

### Can Hydra reproduce this build failure?

Hydra times out on https://hydra.nixos.org/search?query=playwright-browsers (https://github.com/NixOS/hydra/issues/901)

### Link to Hydra build job

_No response_

### Relevant log output

```console
$ nix build .#playwright-driver.browsers
warning: Git tree '/home/leo/nixpkgs' is dirty
error: builder for '/nix/store/ismn0nimsjw77yca3bds3ibnxz5s8nf6-libjxl-0.8.2.drv' failed with exit code 2;
       last 25 log lines:
       > /nix/store/x7w0rmkaam7aikclyhwdbn8wmmq29ma6-binutils-2.44/bin/ld: libjxl.a(frame_header.cc.o):(.rodata+0x4): multiple definition of `jxl::YCbCrChromaSubsampling::kHShift'; CMakeFiles/encode_test.dir/jxl/encode_test.cc.o:(.rodata._ZN3jxl22YCbCrChromaSubsampling7kHShiftE[_ZN3jxl22YCbCrChromaSubsampling7kHShiftE]+0x0): first defined here
       > /nix/store/x7w0rmkaam7aikclyhwdbn8wmmq29ma6-binutils-2.44/bin/ld: libjxl.a(frame_header.cc.o):(.rodata+0x0): multiple definition of `jxl::YCbCrChromaSubsampling::kVShift'; CMakeFiles/encode_test.dir/jxl/encode_test.cc.o:(.rodata._ZN3jxl22YCbCrChromaSubsampling7kVShiftE[_ZN3jxl22YCbCrChromaSubsampling7kVShiftE]+0x0): first defined here
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [lib/CMakeFiles/encode_test.dir/build.make:124: lib/tests/encode_test] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:3147: lib/CMakeFiles/encode_test.dir/all] Error 2
       > [ 82%] Linking CXX executable tests/fast_math_test
       > [ 82%] Built target fast_math_test
       > [ 83%] Linking CXX executable tests/decode_test
       > /nix/store/x7w0rmkaam7aikclyhwdbn8wmmq29ma6-binutils-2.44/bin/ld: libjxl.a(frame_header.cc.o):(.rodata+0x4): multiple definition of `jxl::YCbCrChromaSubsampling::kHShift'; CMakeFiles/decode_test.dir/jxl/decode_test.cc.o:(.rodata._ZN3jxl22YCbCrChromaSubsampling7kHShiftE[_ZN3jxl22YCbCrChromaSubsampling7kHShiftE]+0x0): first defined here
       > /nix/store/x7w0rmkaam7aikclyhwdbn8wmmq29ma6-binutils-2.44/bin/ld: libjxl.a(frame_header.cc.o):(.rodata+0x0): multiple definition of `jxl::YCbCrChromaSubsampling::kVShift'; CMakeFiles/decode_test.dir/jxl/decode_test.cc.o:(.rodata._ZN3jxl22YCbCrChromaSubsampling7kVShiftE[_ZN3jxl22YCbCrChromaSubsampling7kVShiftE]+0x0): first defined here
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [lib/CMakeFiles/decode_test.dir/build.make:124: lib/tests/decode_test] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:3033: lib/CMakeFiles/decode_test.dir/all] Error 2
       > [ 83%] Linking CXX executable tests/fast_dct_test
       > [ 83%] Built target fast_dct_test
       > [ 83%] Linking CXX executable tests/dct_test
       > /nix/store/x7w0rmkaam7aikclyhwdbn8wmmq29ma6-binutils-2.44/bin/ld: libjxl.a(dct_scales.cc.o):(.rodata+0x380): multiple definition of `jxl::WcMultipliers<32ul>::kMultipliers'; CMakeFiles/dct_test.dir/jxl/dct_test.cc.o:(.rodata._ZN3jxl13WcMultipliersILm32EE12kMultipliersE[_ZN3jxl13WcMultipliersILm32EE12kMultipliersE]+0x0): first defined here
       > /nix/store/x7w0rmkaam7aikclyhwdbn8wmmq29ma6-binutils-2.44/bin/ld: libjxl.a(dct_scales.cc.o):(.rodata+0x3c0): multiple definition of `jxl::WcMultipliers<16ul>::kMultipliers'; CMakeFiles/dct_test.dir/jxl/dct_test.cc.o:(.rodata._ZN3jxl13WcMultipliersILm16EE12kMultipliersE[_ZN3jxl13WcMultipliersILm16EE12kMultipliersE]+0x0): first defined here
       > /nix/store/x7w0rmkaam7aikclyhwdbn8wmmq29ma6-binutils-2.44/bin/ld: libjxl.a(dct_scales.cc.o):(.rodata+0x3e0): multiple definition of `jxl::WcMultipliers<8ul>::kMultipliers'; CMakeFiles/dct_test.dir/jxl/dct_test.cc.o:(.rodata._ZN3jxl13WcMultipliersILm8EE12kMultipliersE[_ZN3jxl13WcMultipliersILm8EE12kMultipliersE]+0x0): first defined here
       > /nix/store/x7w0rmkaam7aikclyhwdbn8wmmq29ma6-binutils-2.44/bin/ld: libjxl.a(frame_header.cc.o):(.rodata+0x4): multiple definition of `jxl::YCbCrChromaSubsampling::kHShift'; CMakeFiles/dct_test.dir/jxl/dct_test.cc.o:(.rodata._ZN3jxl22YCbCrChromaSubsampling7kHShiftE[_ZN3jxl22YCbCrChromaSubsampling7kHShiftE]+0x0): first defined here
       > /nix/store/x7w0rmkaam7aikclyhwdbn8wmmq29ma6-binutils-2.44/bin/ld: libjxl.a(frame_header.cc.o):(.rodata+0x0): multiple definition of `jxl::YCbCrChromaSubsampling::kVShift'; CMakeFiles/dct_test.dir/jxl/dct_test.cc.o:(.rodata._ZN3jxl22YCbCrChromaSubsampling7kVShiftE[_ZN3jxl22YCbCrChromaSubsampling7kVShiftE]+0x0): first defined here
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [lib/CMakeFiles/dct_test.dir/build.make:124: lib/tests/dct_test] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:2995: lib/CMakeFiles/dct_test.dir/all] Error 2
       > make: *** [Makefile:146: all] Error 2
       For full logs, run:
         nix log /nix/store/ismn0nimsjw77yca3bds3ibnxz5s8nf6-libjxl-0.8.2.drv
error: 1 dependencies of derivation '/nix/store/2k7ks067gdady7jqq8khdv64pqjkmz43-playwright-webkit.drv' failed to build
error: 1 dependencies of derivation '/nix/store/z0g90qayl557jqwdk7x53jiy29pasawd-playwright-browsers.drv' failed to build
```

### Additional context

_No response_

### System metadata

Run 82f8090f5547aca09fec694a0af6603ef638b746

### Notify maintainers

@kalekseev @nycodeghg @nh2 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/pl/playwright-mcp/package.nix
+++ b/pkgs/by-name/pl/playwright-mcp/package.nix
@@ -8,16 +8,16 @@
 
 buildNpmPackage rec {
   pname = "playwright-mcp";
-  version = "0.0.29";
+  version = "0.0.31";
 
   src = fetchFromGitHub {
     owner = "Microsoft";
     repo = "playwright-mcp";
     tag = "v${version}";
-    hash = "sha256-owSoE3+jSg09dFpM5wv7FJovzsX5ZMp/9IIQhkmSZt0=";
+    hash = "sha256-Hw4OUZCHoquX6Ixv7GlsHcKxqOdJEQYfuDPzqYkVNAk=";
   };
 
-  npmDepsHash = "sha256-jweIBhlVci8CFBIYlFp0opc1ilWMcHt0is4qgTiYNcQ=";
+  npmDepsHash = "sha256-70/t/mgSBwMv9C3VusbjIMMyy3e3npxQLXqKbdL9xa4=";
 
   postInstall = ''
     rm -r $out/lib/node_modules/@playwright/mcp/node_modules/playwright

--- a/pkgs/development/python-modules/playwright/default.nix
+++ b/pkgs/development/python-modules/playwright/default.nix
@@ -22,7 +22,7 @@ in
 buildPythonPackage rec {
   pname = "playwright";
   # run ./pkgs/development/python-modules/playwright/update.sh to update
-  version = "1.53.0";
+  version = "1.54.0";
   pyproject = true;
   disabled = pythonOlder "3.9";
 
@@ -30,7 +30,7 @@ buildPythonPackage rec {
     owner = "microsoft";
     repo = "playwright-python";
     tag = "v${version}";
-    hash = "sha256-jFS2Luq/9mRsXZ65H3VLw+sTBplVNVy/yZYrpF5Hc0M=";
+    hash = "sha256-xyuofDL0hWL8Gn4sYNLKte8q/4bMo+3aSbYaf5iWiBk=";
   };
 
   patches = [

--- a/pkgs/development/python-modules/playwright/update.sh
+++ b/pkgs/development/python-modules/playwright/update.sh
@@ -22,6 +22,19 @@ repo_url_prefix="https://github.com/microsoft/playwright/raw"
 temp_dir=$(mktemp -d)
 trap 'rm -rf "$temp_dir"' EXIT
 
+# Update playwright-mcp package
+mcp_version=$(curl ${GITHUB_TOKEN:+" -u \":$GITHUB_TOKEN\""} -s https://api.github.com/repos/microsoft/playwright-mcp/releases/latest | jq -r '.tag_name | sub("^v"; "")')
+update-source-version playwright-mcp "$mcp_version"
+
+# Update npmDepsHash for playwright-mcp
+pushd "$temp_dir" >/dev/null
+curl -fsSL -o package-lock.json "https://raw.githubusercontent.com/microsoft/playwright-mcp/v${mcp_version}/package-lock.json"
+mcp_npm_hash=$(prefetch-npm-deps package-lock.json)
+rm -f package-lock.json
+popd >/dev/null
+
+mcp_package_file="$root/../../../by-name/pl/playwright-mcp/package.nix"
+sed -E 's#\bnpmDepsHash = ".*?"#npmDepsHash = "'"$mcp_npm_hash"'"#' -i "$mcp_package_file"
 
 
 # update binaries of browsers, used by playwright.
@@ -47,6 +60,9 @@ update_browser() {
         fi
     else
         if [ "$name" = "ffmpeg" ] || [ "$name" = "chromium-headless-shell" ]; then
+            suffix="linux"
+        elif [ "$name" = "chromium" ]; then
+            stripRoot="true"
             suffix="linux"
         elif [ "$name" = "firefox" ]; then
             stripRoot="true"
@@ -81,7 +97,7 @@ curl -fsSl \
       )
     ' > "$playwright_dir/browsers.json"
 
-# We currently use Chromium from nixpkgs, so we don't need to download it here
+update_browser "chromium" "linux"
 update_browser "chromium-headless-shell" "linux"
 update_browser "firefox" "linux"
 update_browser "webkit" "linux"

--- a/pkgs/development/web/playwright/browsers.json
+++ b/pkgs/development/web/playwright/browsers.json
@@ -2,19 +2,19 @@
   "comment": "This file is kept up to date via update.sh",
   "browsers": {
     "chromium": {
-      "revision": "1179",
-      "browserVersion": "138.0.7204.23"
+      "revision": "1181",
+      "browserVersion": "139.0.7258.5"
     },
     "chromium-headless-shell": {
-      "revision": "1179",
-      "browserVersion": "138.0.7204.23"
+      "revision": "1181",
+      "browserVersion": "139.0.7258.5"
     },
     "firefox": {
-      "revision": "1488",
-      "browserVersion": "139.0"
+      "revision": "1489",
+      "browserVersion": "140.0.2"
     },
     "webkit": {
-      "revision": "2182",
+      "revision": "2191",
       "revisionOverrides": {
         "debian11-x64": "2105",
         "debian11-arm64": "2105",
@@ -29,7 +29,7 @@
         "ubuntu20.04-x64": "2092",
         "ubuntu20.04-arm64": "2092"
       },
-      "browserVersion": "18.5"
+      "browserVersion": "26.0"
     },
     "ffmpeg": {
       "revision": "1011",

--- a/pkgs/development/web/playwright/chromium-headless-shell.nix
+++ b/pkgs/development/web/playwright/chromium-headless-shell.nix
@@ -30,8 +30,8 @@ let
       stripRoot = false;
       hash =
         {
-          x86_64-linux = "sha256-C545VC0RxFYfKf8XArfVoI2yzrGyfR7vxjryJHfcfBQ=";
-          aarch64-linux = "sha256-i+HVf/6Qz2nCLLzPxbRYVFjDDOhd5ETYuHje5YsmsAo=";
+          x86_64-linux = "sha256-AYh2urKZdjXCELimYaFihWp0FbDLf4uRrKLJZVxug5M=";
+          aarch64-linux = "sha256-diBiy0z51BxGK0PcfQOf1aryUcZesKu/UHBSZUjqwMk=";
         }
         .${system} or throwSystem;
     };
@@ -66,8 +66,8 @@ let
     stripRoot = false;
     hash =
       {
-        x86_64-darwin = "sha256-2pe1LS3WXRG/V2k/BNN1MmOOdGoA0WCFhUpZW8TUhic=";
-        aarch64-darwin = "sha256-U07QnHFPQXyO3VGWfZvcP8cJEmVJAJ7imj+6DQlN9vQ=";
+        x86_64-darwin = "sha256-vIJuDjkasUYlMW0aCOyztyrlh5kvcwNR9GBaoa/yh/M=";
+        aarch64-darwin = "sha256-6Q6nz0H2749srdMF/puk/gnG1gQBEnWe9cQO3owL2OU=";
       }
       .${system} or throwSystem;
   };

--- a/pkgs/development/web/playwright/chromium.nix
+++ b/pkgs/development/web/playwright/chromium.nix
@@ -41,8 +41,8 @@ let
       url = "https://playwright.azureedge.net/builds/chromium/${revision}/chromium-${suffix}.zip";
       hash =
         {
-          x86_64-linux = "sha256-7oQQCAIt1VJiMNFEJO40K8oENK/L0BICXm2D/3fZ8bA=";
-          aarch64-linux = "sha256-1OmByLX2jNHXAzWdXF8Od7S7pj/jl4wwvOQcsZc5R7o=";
+          x86_64-linux = "sha256-R7nMCVpUqgRwtB0syhfIK81maiTVWr8lYBLp4bR8VBg=";
+          aarch64-linux = "sha256-4fc4X7QwBigktmEeseuqIyEeV70Dy3eO/femXrftMd0=";
         }
         .${system} or throwSystem;
     };
@@ -109,8 +109,8 @@ let
     stripRoot = false;
     hash =
       {
-        x86_64-darwin = "sha256-KOoCbygsZZzGNKD8ICcGg0iM2h0HVgXq0I4JMPaUJR8=";
-        aarch64-darwin = "sha256-2naFzKWmo6el+AqljzILO+hUq/E2g81Dt1fwq79EYO8=";
+        x86_64-darwin = "sha256-0u1AStbUTX+qgUmg2DvL59B4b265WywDaBV+MdSuaNE=";
+        aarch64-darwin = "sha256-4pg4wmNTF8mw+APmdpvYlFxb9zc6OUh11oW5gCRKETY=";
       }
       .${system} or throwSystem;
   };

--- a/pkgs/development/web/playwright/driver.nix
+++ b/pkgs/development/web/playwright/driver.nix
@@ -27,13 +27,13 @@ let
     }
     .${system} or throwSystem;
 
-  version = "1.53.1";
+  version = "1.54.1";
 
   src = fetchFromGitHub {
     owner = "Microsoft";
     repo = "playwright";
     rev = "v${version}";
-    hash = "sha256-N5BS8zpoQGUf5gly0fyutaK76CAhbwOGAUofGnfkmnM=";
+    hash = "sha256-xwyREgelHLkpbUXOZTppKK7L6dE4jx0d/lbDWSKGzTY=";
   };
 
   babel-bundle = buildNpmPackage {
@@ -70,7 +70,7 @@ let
     pname = "utils-bundle-core";
     inherit version src;
     sourceRoot = "${src.name}/packages/playwright-core/bundles/utils";
-    npmDepsHash = "sha256-3hdOmvs/IGAgW7vhldms9Q9/ZQfbjbc+xP+JEtGJ7g8=";
+    npmDepsHash = "sha256-gEm2oTxj4QIiGnIOPffOLh3BYSngpGToF89ObnDYBqs=";
     dontNpmBuild = true;
     installPhase = ''
       cp -r . "$out"
@@ -92,7 +92,7 @@ let
     inherit version src;
 
     sourceRoot = "${src.name}"; # update.sh depends on sourceRoot presence
-    npmDepsHash = "sha256-a1s1l8PG0ViVqYOksB2dkID/AHczMjLNQJW88+yB0B0=";
+    npmDepsHash = "sha256-4bsX8Q8V3CBpIsyqMYTzfERQQPY5zlPf7CoqR6UkUHU=";
 
     nativeBuildInputs = [
       cacert

--- a/pkgs/development/web/playwright/firefox.nix
+++ b/pkgs/development/web/playwright/firefox.nix
@@ -17,8 +17,8 @@ let
       }.zip";
       hash =
         {
-          x86_64-linux = "sha256-L9bIldFCqZ/jnNKkJk6nS0HNaJefzTMQIJ6VLUE9ugc=";
-          aarch64-linux = "sha256-iuiS59f8j3K+grBU7ZtZPfU4r2Dp7s0JJHf2n/4r30U=";
+          x86_64-linux = "sha256-j7gOuXMyftNQencgfpk8Y4ED2LuT7TAa30IPyzmir48=";
+          aarch64-linux = "sha256-deIUGKBrp56TsDr61cbNbRRSRcVpSoa6pdmMk4oB/Eg=";
         }
         .${system} or throwSystem;
     };
@@ -41,8 +41,8 @@ let
     stripRoot = false;
     hash =
       {
-        x86_64-darwin = "sha256-K0eW1kC1tckJu0crD89hDhK8PHyQUB0YUYN9DdX0HKw=";
-        aarch64-darwin = "sha256-n1Uy59r6wxmung8QKvw3JeyF3ec/avCVp9fI+bck/iA=";
+        x86_64-darwin = "sha256-ljgFoyqCg9kma2cDFodNjbkAeEylIzVdWkS1vU/9Rbg=";
+        aarch64-darwin = "sha256-W2J5APPWEkmoDgBEox6/ygg2xyWpOHZESXFG0tZbj1M=";
       }
       .${system} or throwSystem;
   };

--- a/pkgs/development/web/playwright/webkit.nix
+++ b/pkgs/development/web/playwright/webkit.nix
@@ -102,7 +102,11 @@ let
           hash = "sha256-X4fbYTMS+kHfZRbeGzSdBW5jQKw8UN44FEyFRUtw0qo=";
         })
       ];
-      postPatch = "";
+      postPatch = ''
+        # Fix multiple definition errors by using C++17 instead of C++11
+        substituteInPlace CMakeLists.txt \
+          --replace "set(CMAKE_CXX_STANDARD 11)" "set(CMAKE_CXX_STANDARD 17)"
+      '';
       postInstall = "";
 
       cmakeFlags =

--- a/pkgs/development/web/playwright/webkit.nix
+++ b/pkgs/development/web/playwright/webkit.nix
@@ -130,8 +130,8 @@ let
       stripRoot = false;
       hash =
         {
-          x86_64-linux = "sha256-lwH783B3/laqw0IdGBnVzvySRoF0AwZsSolaqUKmsM4=";
-          aarch64-linux = "sha256-qtvP0bc5rcZcz6SqigfdrjhTWEmvT4k11I1GW1Eoj/Q=";
+          x86_64-linux = "sha256-OSVHFGdcQrzmhLPdXF61tKmip/6/D+uaQgSBBQiOIZI=";
+          aarch64-linux = "sha256-b8XwVMCwSbujyqgkJKIPAVNX83Qmmsthprr2x9XSb10=";
         }
         .${system} or throwSystem;
     };
@@ -211,8 +211,8 @@ let
     stripRoot = false;
     hash =
       {
-        x86_64-darwin = "sha256-p1+Pk+Zhf2OPEmEWCEd0tA7CdoMcOgYp69SnQXufFJ0=";
-        aarch64-darwin = "sha256-tEfKvJuGe4htZLSOn94eKeBtWXYkjl73iJSY4BWJMKo=";
+        x86_64-darwin = "sha256-shjhozJS2VbBjpjJVlM9hwBzGWwgva1qhfEUhY8t9Bk=";
+        aarch64-darwin = "sha256-ZRl86L/OOTNPWfZDl6JQfuXL41kI/Wir99/JIbf7T7M=";
       }
       .${system} or throwSystem;
   };

